### PR TITLE
Fix for multiple engine run

### PR DESCRIPTION
**Pull request by benedikt-voelkel**
* gtrak/gtreveroot.F, geant321/sckine.inc
  Single run: Preserve previous logic exactly for indexing and
              transport.
  Multi run: Explicitly rely on user information whether a primary or
             secondary was popped. No implicit assumptions on track
             indexing.

* TGeant3/TGeant3.cxx
  Single run: Preserve previous logic exactly when popping tracks.
  Multi run: Extract if primary based on parent ID of TMCParticleStatus.
             Current track number is now set implicitly when
             TMCManagerStack::PopNextTrack(...) is called.

### CODE REVIEW

File: TGeant3/TGeant3.cxx [hunk 1]
```diff
@@ -6809,7 +6827,8 @@ extern "C" void type_of_call rxouth()
    //
    // Called by Gtreve at the end of each primary track
    //
-   if (gDoPostTrackHooks && gDoPrimaryHooks) {
+   if (gDoFinishPrimaryHook) {
+      std::cout << "FINISH" << std::endl;
```
Comment (ihrivnac) [inline TGeant3/TGeant3.cxx]: Can you, please, remove this debug printing?

File: TGeant3/TGeant3.cxx [hunk 2]
```diff
@@ -6820,7 +6839,8 @@ extern "C" void type_of_call rxinh()
    //
    // Called by Gtreve at the beginning of each primary track
    //
-   if (gDoPreTrackHooks && gDoPrimaryHooks) {
+   if (gDoPrimaryHooks && gDoPreTrackHooks) {
+      std::cout << "BEGIN" << std::endl;
```
Comment (ihrivnac) [inline TGeant3/TGeant3.cxx]: And also this one?

**Comment by benedikt-voelkel** [comment]
Hello @ihrivnac 
Printing has been removed and there is now the updated version.

**Comment by ihrivnac** [comment]
Thank you.

**Pull request merged**